cinema/migrations/0002_actor_cinemahall_genre_movie_actors_movie_genres.py

```diff
@@ -0,0 +1,47 @@
+# Generated by Django 4.1 on 2024-12-22 17:50
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('cinema', '0001_initial'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='Actor',
+            fields=[
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('first_name', models.CharField(max_length=50)),
+                ('last_name', models.CharField(max_length=50)),
+            ],
+        ),
+        migrations.CreateModel(
+            name='CinemaHall',
+            fields=[
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('name', models.CharField(max_length=100)),
+                ('rows', models.PositiveIntegerField()),
+                ('seats_in_row', models.PositiveIntegerField()),
+            ],
+        ),
+        migrations.CreateModel(
+            name='Genre',
+            fields=[
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('name', models.CharField(max_length=100, unique=True)),
+            ],
+        ),
+        migrations.AddField(
+            model_name='movie',
+            name='actors',
+            field=models.ManyToManyField(to='cinema.actor'),
+        ),
+        migrations.AddField(
+            model_name='movie',
+            name='genres',
+            field=models.ManyToManyField(to='cinema.genre'),
```

Comment:
The `ManyToManyField` for `genres` in the `Movie` model should include a `related_name` attribute as per the task requirements. This allows for reverse querying from the `Genre` model.